django/website/logframe/views.py

```diff
@@ -78,13 +77,21 @@ class CreateLogframe(ManageLogFrame, CreateView):
     form_class = CreateLogFrameForm
     template_name = 'logframe/create_logframe.html'
 
+    def add_result_levels(self):
+        ResultLevelName.objects.bulk_create([
+            ResultLevelName(level_number=1, level_name='Goal', logframe=self.object),
+            ResultLevelName(level_number=2, level_name='Outcome', logframe=self.object),
+            ResultLevelName(level_number=3, level_name='Output', logframe=self.object),
+            ResultLevelName(level_number=4, level_name='Activity', logframe=self.object)
+        ])
 
     def form_valid(self, form):
         # The logframe needs to exist before we create the settings, so finish
         # the regular form_valid code.
         form.instance.organization = Organization.objects.get(slug=self.kwargs['org_slug'])
         response = CreateView.form_valid(self, form)
         Settings.objects.create(logframe=form.instance)
+        self.add_result_levels()
```

Comment:
I guess this is consistent with creating the Settings in the line above. Is it better practice to do all this related object creation in post_save so that it happens outside of the form?